src/seedu/addressbook/commands/EditCommand.java

```diff
@@ -0,0 +1,47 @@
+package seedu.addressbook.commands;
+
+import seedu.addressbook.common.Messages;
+import seedu.addressbook.data.person.Person;
+import seedu.addressbook.data.person.ReadOnlyPerson;
+import seedu.addressbook.data.person.UniquePersonList.PersonNotFoundException;
+
+/**
+ * Edits a person's name identified using it's last displayed index from the address book.
+ */
+
+public class EditCommand extends Command {
+    public static final String COMMAND_WORD = "edit";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD
+            + ": Edits the person's name identified by the index number used in the last person listing.\n"
+            + "Parameters: INDEX\n"
+            + "Example: " + COMMAND_WORD + " 1" + " Susan";
+
+    public static final String MESSAGE_EDIT_PERSON_SUCCESS = "Edited Person: %1$s";
+
+    private final int indexToEdit;
+    private final String nameToEdit;
+
+
+    public EditCommand(int index, String name) {
+        this.indexToEdit = index;
+        this.nameToEdit = name;
+    }
+
+
+    @Override
+    public CommandResult execute() {
+        try {
+            setTargetIndex(this.indexToEdit);
+            final ReadOnlyPerson target = getTargetPerson();
+            addressBook.editName(this.indexToEdit, this.nameToEdit);
+            return new CommandResult(String.format(MESSAGE_EDIT_PERSON_SUCCESS, target));
+
+        } catch (IndexOutOfBoundsException ie) {
+            return new CommandResult(Messages.MESSAGE_INVALID_PERSON_DISPLAYED_INDEX);
+        } catch (PersonNotFoundException pnfe) {
+            return new CommandResult(Messages.MESSAGE_PERSON_NOT_IN_ADDRESSBOOK);
+        }
+    }
```

Comment:
Doesn't seem like the editing method checks for any duplicate name replacement, e.g. someone might already be called Susan in the address book while you choose to edit someone else into Susan as well. Not sure if this seems contradictory to how the current address book doesn't allow addition of duplicated persons. (correct me if I'm mistaken!)

Reply:
Yes this is actually a violation of no duplicate persons as specified in add command. In this case, you need to check if the new name is the same as any existing names in the address book. Usually when we edit/update the details of a person, we only edit the phone/email/address instead of the name.